src/exercise/02.js

```diff
@@ -4,29 +4,22 @@
 import * as React from 'react'
 import {Switch} from '../switch'
 
-function Toggle() {
+function Toggle({ children }) {
   const [on, setOn] = React.useState(false)
   const toggle = () => setOn(!on)
-
-  // 🐨 replace this with a call to React.Children.map and map each child in
-  // props.children to a clone of that child with the props they need using
-  // React.cloneElement.
-  // 💰 React.Children.map(props.children, child => {/* return child clone here */})
-  // 📜 https://reactjs.org/docs/react-api.html#reactchildren
-  // 📜 https://reactjs.org/docs/react-api.html#cloneelement
-  return <Switch on={on} onClick={toggle} />
+  return React.Children.map(children, (child) => {
+    return React.cloneElement(child, {
+      on,
+      toggle,
+    })
+  })
 }
 
-// 🐨 Flesh out each of these components
-
-// Accepts `on` and `children` props and returns `children` if `on` is true
-const ToggleOn = () => null
+const ToggleOn = ({ children, on }) => on ? children : null
 
-// Accepts `on` and `children` props and returns `children` if `on` is false
-const ToggleOff = () => null
+const ToggleOff = ({ children, on }) => (!on) ? children : null
```

Comment:
Parenthesis aren't necessary here:
```suggestion
const ToggleOff = ({ children, on }) => !on ? children : null
```